storage/index.js

```diff
@@ -1,7 +1,45 @@
+var createStorageBackend = require('./backend')
+
 /**
  * Storage wrapper object for relaying operations to backend.
  */
-module.exports = function (backend) {
+module.exports = function (STATE, URL) {
+  var backend = undefined
+  var backendCache = {}
+
+  // TODO add some clean up functionality in update for cached backends
+  // that no longer should stay alive to clean up resources etc
+
+  function createNewBackend (state, url) {
+    var debug = state.params.debug
+    backend = createStorageBackend(state, url)
+    if (backend) {
+      if (debug) console.log('storage: created new backend for url', url)
+      backendCache[url] = backend
+    } else {
+      console.warn('storage: missing protocol handler for url', url)
+    }
+  }
+
+  function updateBackend (state, url) {
+    var debug = state.params.debug
+    if (backend) {
+      var cached = backendCache[url]
+      if (cached && cached === backend) {
+        if (debug) console.log('storage: not updating backend')
+      } else if (cached) {
+        if (debug) console.log('storage: updating to cached backend for url', url)
```

Comment:
There's a cached backend, but it's not us, so we just switch to it. This avoids having to re-create the storage, with setup code for hypercore etc.